### DISCUSSION TITLE
Utilisation of ports x 10

### DISCUSSION THREAD

**Discussion by cooper-fm** [AzuraCast/AzuraCast#7489] [General]
Hi chaps,

As we've been scaling up the number of stations within our AC server, it's becoming apparent that AC likes to reserve 10 ports at a time when creating radio stations.

Is there anyway to get around this? We're looking to create 150 stations initially, potentially up to 500 (and maybe even more later).  I'd like to have the port range more compact if possible.

Thanks

**Comment by Heimatradio** [comment]
run the AzuraCast Setup with
./docker.sh install
and change the port-range to eg. 8000-15999
BUT, what kind of huge server should 500 stations run on???

**Reply by cooper-fm** [reply]
Sure, we get we can extend the range, but was hoping to compact the port range and not waste ports.

We're "rebooting" an old radio station project which had around 250 stations (150 stations with traffic worthwhile transferring), but as our business model is evolving there may easily be the potential that we'll go up to around 500 stations of in-house content. Offering our services to selected partners, this could potentially move up to 1,000 +, but at that stage we'll just build a server cluster. But that's a thought for next year.

But surely 500 stations is not unheard of, are there not resellers that use AC?

_See my earlier thread concerning re-encoding, and you'll see we're trying to make this as scalable and as efficient as possible :)_

**Comment by BusterNeece** [comment]
Currently, there is no way to change the "gap" we allocate between station broadcasting ports. This is to allow us to use some of those ports for the internal back-and-forth communications that happen between the station processes and AzuraCast, incoming DJ connections, and to reserve some space for potential future use. It would be much, much harder for us to expand this port clearance in the future if we needed it than to just have that extra space allocated for our purposes.

Notably, we don't clear the *entire* range through the firewall, just the external-facing ports used in that range. That does dramatically reduce the number of ports proxied through to Docker. We automatically generate the assigned range based on the port-range minimum/maximum specified in the environment variables.

What I will say is that 500 stations will very, very likely not run at all, no matter how much hardware you throw at it. AzuraCast isn't built for that level of concurrency; we've improved this significantly since launch and over the last few years, but there are just things that don't parallelize that well across that many stations with a single database, web UI, etc. running. The experience this offers customers would be likely suboptimal.
